### DISCUSSION TITLE
How to return back to the interactive shell directly from any function on the stack

### DISCUSSION THREAD

**Discussion by MidAutumnMoon** [fish-shell/fish-shell#8878] [Q&A] [answered]
My question is, in a scenario that a `parent` function called another `child` function, where that `parent` is called on the interactive shell, is it possible to return directly back to our interactive shell from that `child` function?

The current solution is to use `return` with `and/or` checks manually after each call to `child`, which is not elegant in my opinion. I want something like `exit` but without causing the fish process to shutdown.

**Comment by ridiculousfish** [comment, marked as answer]
The only thing approximating non-local control flow is signal handling. This "sort of" works:

```
function child
	echo "Child ran"
	kill -INT $fish_pid 
	echo "This will not run"
end

function parent
	echo "Parent ran"
	child
	echo "This will not run either"
end

parent
```

and will restore interactive control. It runs into problems if the child function is in a pipeline, etc. but maybe this is good enough for your use case.

**Reply by MidAutumnMoon** [reply]
Thanks! I will take this tricky method into consideration.

**Comment by floam** [comment]
There's `breakpoint` - if this was for some kind of "something went wrong" situation - this is specifically for just getting a chance to change things before continuing to the control flow (or return early).
